backend/src/main/java/ca/gov/dtsstn/vacman/api/service/MicrosoftGraphService.java

```
@@ -0,0 +1,130 @@
+     * @return MicrosoftGraphUserModel containing user information
+     * @throws IllegalStateException if no authentication context is available
+     * @throws RestClientException if the Microsoft Graph API call fails
+     */
+    public MicrosoftGraphUserModel getCurrentUser() {
+        log.debug("Retrieving current user information from Microsoft Graph");
+
+        // Extract the bearer token from Spring Security context
+        String bearerToken = extractBearerTokenFromContext();
+
+        // Prepare headers with authorization
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.setBearerAuth(bearerToken);
+        headers.set("ConsistencyLevel", "eventual"); // For potential advanced queries
+
+        HttpEntity<String> entity = new HttpEntity<>(headers);
+
+        try {
+            // Make the request to Microsoft Graph
+            String url = properties.baseUrl() + USER_PROFILE_ENDPOINT;
+            log.debug("Making request to Microsoft Graph: {}", url);
+
+            ResponseEntity<MicrosoftGraphUserModel> response = restTemplate.exchange(
+                url,
+                HttpMethod.GET,
+                entity,
+                MicrosoftGraphUserModel.class
+            );
+
+            MicrosoftGraphUserModel user = response.getBody();
+            if (user != null) {
+                log.info("Successfully retrieved user information for user ID: {}", user.id());
+                return user;
+            } else {
+                log.warn("Microsoft Graph returned null response body");
+                throw new RestClientException("Microsoft Graph returned empty response");
+            }
+
+        } catch (RestClientException e) {
+            log.error("Failed to retrieve user information from Microsoft Graph", e);
+            throw new RestClientException("Failed to retrieve user information from Microsoft Graph: " + e.getMessage(), e);
+        }
```

I don't think this exception handling is overly useful. Just make the call and if it throws, let the exception bubble up to the global exception handler where it will be logged.